lxml-stubs/etree.pyi

```diff
@@ -250,34 +262,36 @@ class _ElementTree:
         self,
         source: _FileSource,
         parser: Optional[_AnyParser] = ...,
-        base_url: Optional[_AnyStr] = ...,
+        base_url: Optional[_StrOrBytes] = ...,
     ) -> _Element: ...
     def write(
         self,
         file: _FileSource,
-        encoding: _AnyStr = ...,
-        method: _AnyStr = ...,
+        encoding: Optional[_StrOrBytes] = ...,
```

Comment:
This should be marked as Optional since None is allowed.

Reply:
Unlike the global `tostring` function this only accepts unicode or bytes.

```python
elm_tree = etree.fromstring("<test/>").getroottree()
print(type(elm_tree))

with open("test.xml", "wb") as out_file:
   elm_tree.write(out_file, encoding = str)
```

```
Traceback (most recent call last):
  File "C:\Source\p5\taccs\unicode_xml.py", line 82, in <module>
    elm_tree.write(out_file, encoding = str)
  File "src/lxml/etree.pyx", line 2055, in lxml.etree._ElementTree.write
TypeError: unbound method str.upper() needs an argument
```

Reply:
Then let's start the deprecation by making sure users pass only strings.